### PR TITLE
Tidying codes WIP

### PR DESCRIPTION
**Pull request by jasmin-guven**
Created PCN.py class, functions.py and main.py. 

Plotted graphs on plot_link_lengths.ipynb. Plots look good, but need to shift the theory plots down.

### CODE REVIEW

File: src/PCN.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		Return
+		------
+		chain_length: int
+			length of the chain
+		'''
+		chain_length = len(C_alphas)
+		
+		return chain_length
+
+	def create_connected_component_subgraphs(self,protein_graph):
```
Comment (ppxasjsm) [inline src/PCN.py]: make sure this is formatted in the same way self, protein_graph. I would recommend getting something like pycharm and let it autoformat everything according to pep8

File: src/PCN.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+Base class for PCNs. 
+
+'''
+import networkx as nx
+import MDAnalysis as mda
+import numpy as np
+import random
+
+class PCN():
+	'''
```
Comment (ppxasjsm) [inline src/PCN.py]: Do you use tabs? Or 8 spaces? typically in Python we want to use 4 single white spaces

File: src/PCN.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+	Base class for PCNs. 
+	
+	Takes a PDB file and creates a PCN from the alpha carbons.
+
+	'''
+	# Define a threshold value for defining a link between two atoms
+	link_length_threshold = 8.0 # Å
```
Comment (ppxasjsm) [inline src/PCN.py]: How can you change this variable? maybe look at decorators getters and setters

**Review by ppxasjsm — COMMENTED** [review]
This looks much better in general! clear and easy to read now. Happy for this to be merged in principle, but you might look at autoformatting around pep 8 using something like pycharm.

File: src/PCN.py (hunk 1 above)
Comment (jasmin-guven) [reply src/PCN.py]: Yep fixed.

File: src/PCN.py (hunk 2 above)
Comment (jasmin-guven) [reply src/PCN.py]: I'm using sublime text for python so the spacing is automatic with tabs which is why GitHub reads this as 8 spaces. I've fixed the settings now.

**Pull request closed without merging**